packages/material-ui/src/Button/Button.js

```diff
@@ -158,6 +158,25 @@ export const styles = theme => ({
       },
     },
   },
+  /* Styles applied to the root element if `disableElevation={true}`. */
+  disableElevation: {
+    boxShadow: 'none',
+    '&:hover': {
+      boxShadow: 'none',
+      '@media (hover: none)': {
```

Comment:
I suspect we can kill this style, media queries aren't supposed to increase specificity.